contracts/token/ERC20/extensions/ERC20uRWA.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import {IERC7943} from "../../../interfaces/IERC7943.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {ERC165, IERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ERC20Freezable} from "./ERC20Freezable.sol";
+import {ERC20Restricted} from "./ERC20Restricted.sol";
+
+/**
+ * @dev Extension of {ERC20} according to https://eips.ethereum.org/EIPS/eip-7943[EIP-7943].
+ *
+ * Combines standard ERC-20 functionality with RWA-specific features like user restrictions,
+ * asset freezing, and forced asset transfers.
+ */
+abstract contract ERC20uRWA is ERC20, ERC165, ERC20Freezable, ERC20Restricted, IERC7943 {
+    /// @inheritdoc ERC20Restricted
+    function isUserAllowed(address user) public view virtual override(IERC7943, ERC20Restricted) returns (bool) {
+        return super.isUserAllowed(user);
+    }
+
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165, IERC165) returns (bool) {
+        return interfaceId == type(IERC7943).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /**
+     * @dev See {IERC7943-isTransferAllowed}.
+     *
+     * CAUTION: This function is only meant for external use. Overriding it will not apply the new checks to
+     * the internal {_update} function. Consider overriding {_update} accordingly to keep both functions in sync.
+     */
+    function isTransferAllowed(address from, address to, uint256, uint256 amount) external view virtual returns (bool) {
+        return (amount <= available(from) && isUserAllowed(from) && isUserAllowed(to));
+    }
+
+    /// @inheritdoc IERC7943
+    function getFrozen(address user, uint256) public view virtual returns (uint256 amount) {
+        return frozen(user);
+    }
+
+    /**
+     * @dev See {IERC7943-setFrozen}.
+     *
+     * NOTE: The `amount` is capped to the balance of the `user` to ensure the {IERC7943-Frozen} event
+     * emits values that consistently reflect the actual amount of tokens that are frozen.
+     */
+    function setFrozen(address user, uint256, uint256 amount) public virtual {
+        uint256 actualAmount = Math.min(amount, balanceOf(user));
```

Comment:
https://github.com/ethereum/ERCs/blob/9f76a19660ffec0e57df3ed2d660ec1d03993f6c/ERCS/erc-7943.md?plain=1#L147

The ERC does specify this, but I'm not sure I agree.